src/app/screens/ProfileScreen/component/UpdateSkillModal.tsx

```diff
@@ -0,0 +1,334 @@
+    const {otherSkills} = getValues();
+
+    const skills =
+      otherSkills?.split(',').filter(e => e !== '' && e !== skill) || [];
+
+    setValue('otherSkills', skills.join(','));
+  };
+
+  return (
+    <Modal
+      isVisible={isVisible}
+      animationIn={'slideInUp'}
+      animationOut={'slideOutDown'}
+      animationInTiming={500}
+      animationOutTiming={500}
+      contentStyle={styles.contentStyle}>
+      <ScrollView>
+        <Text style={styles.title}>Update Skills</Text>
+
+        <View style={styles.fieldStyle}>
+          <Text style={styles.labelText}>Primary Technical Skill</Text>
+          <Controller
+            control={control}
+            render={({field: {onChange, value}}) => (
+              <PickerSelect
+                placeholder={{
+                  label: strings.SELECT,
+                  value: '',
+                }}
+                error={errors.primarySkill?.message}
+                disabled={isLoading}
+                onValueChange={onChange}
+                value={value ? value : strings.SELECT}
+                items={skillsList}
+              />
+            )}
+            name="primarySkill"
+          />
+        </View>
+
+        <View style={styles.fieldStyle}>
+          <Text style={styles.labelText}>Secondary Technical Skill</Text>
+          <Controller
+            control={control}
+            render={({field: {onChange, value}}) => (
+              <PickerSelect
+                placeholder={{
+                  label: strings.SELECT,
+                  value: '',
+                }}
+                error={errors.secondarySkill?.message}
+                disabled={isLoading}
+                onValueChange={onChange}
+                value={value ? value : strings.SELECT}
+                items={skillsList}
+              />
+            )}
+            name="secondarySkill"
+          />
+        </View>
+
+        <View style={styles.fieldStyle}>
+          <Text style={styles.labelText}>Ternary Technical Skill</Text>
+          <Controller
+            control={control}
+            render={({field: {onChange, value}}) => (
+              <PickerSelect
+                placeholder={{
+                  label: strings.SELECT,
+                  value: '',
+                }}
+                error={errors.ternarySkill?.message}
+                disabled={isLoading}
+                onValueChange={onChange}
+                value={value ? value : strings.SELECT}
+                items={skillsList}
+              />
+            )}
+            name="ternarySkill"
+          />
+        </View>
+
+        <View style={styles.fieldStyle}>
+          <Text style={styles.labelText}>Other Skills</Text>
+          <Controller
+            control={control}
+            render={({field: {value}}) => (
+              <>
+                <View style={styles.otherSkillChipContainer}>
+                  {value
+                    ?.split(',')
+                    .filter(e => e)
```

Comment:
why do we need filter here

Reply:
In case value is an empty string, split create an array `['']` which will create an empty chip

Reply:
ok